cmd/collector/app/handler/otlp_receiver.go

```diff
@@ -101,32 +99,32 @@ func startOTLPReceiver(
 	return otlpReceiver, nil
 }
 
-func applyGRPCSettings(cfg *configgrpc.ServerConfig, opts *flags.GRPCOptions) {
-	if opts.HostPort != "" {
-		cfg.NetAddr.Endpoint = opts.HostPort
+func applyGRPCSettings(cfg *configgrpc.ServerConfig, opts *configgrpc.ServerConfig) {
```

Comment:
I don't think we need this function anymore. Previously it was performing translation, but now we already have the desired struct that can be used directly.